src/lib/base-dao-mongoose.js

```diff
@@ -1,35 +1,77 @@
+import mongoose from 'mongoose';
+import {
+  DatabaseWrongIDError,
+  ResourceDoesNotExistAnymore,
+  ResourceDuplicateError
+} from '../lib/errors/database';
+
 class BaseDaoMongoose {
   constructor(Model) {
     this.Model = Model;
   }
 
-  create(item) {
+  async create(item) {
     this.checkType(item);
-    return Promise.resolve(item.save());
+    try {
+      return await Promise.resolve(item.save());
+    } catch (error) {
+      if (error.code === 11000) {
+        throw new ResourceDuplicateError();
+      }
+      throw error;
+    }
   }
 
-  getAll() {
+  async getAll() {
     return Promise.resolve(this.Model.find({}));
   }
 
-  getById(id) {
-    return Promise.resolve(this.Model.findById(id));
+  async getById(id) {
+    try {
+      const item = await Promise.resolve(this.Model.findById(id));
+      if (!item) {
+        throw new ResourceDoesNotExistAnymore();
+      }
+      return item;
+    } catch (error) {
+      if (error instanceof mongoose.CastError && error.kind === 'ObjectId') {
+        throw new DatabaseWrongIDError();
```

Comment:
There are too many custom errors. This can easily be a usual Error (500), along many of the others.
Chill and simplify :)